src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -326,6 +326,19 @@ private List<LibraryDependency> GetDependenciesFromSpecRestoreMetadata(PackageSp
                 var targetFrameworkInfo = packageSpec.GetTargetFramework(targetFramework);
                 dependencies.AddRange(targetFrameworkInfo.Dependencies);
 
+                if (packageSpec.RestoreMetadata?.CentralPackageVersionsEnabled == true)
+                {
+                    var dependencyNamesSet = new HashSet<string>(targetFrameworkInfo.Dependencies.Select(d => d.Name), StringComparer.OrdinalIgnoreCase);
+                    dependencies.AddRange(targetFrameworkInfo.CentralPackageVersions
+                        .Where(item => !dependencyNamesSet.Contains(item.Key))
+                        .Select(item => new LibraryDependency()
+                        {
+                            LibraryRange = new LibraryRange(item.Value.Name, item.Value.VersionRange, LibraryDependencyTarget.Package),
+                            VersionCentrallyManaged = true,
+                            ReferenceType = LibraryDependencyReferenceType.None,
```

Comment:
In line 332, you're adding central dependencies that are not already in the `depenencies` list. Are those central dependencies considered transitive ? If that's the case, should the ReferenceType be `LibraryDependencyReferenceType.Transitve` ?

Reply:
No, at this point they are not transitive for this reason set to none.